src/main/java/com/example/concurrency/BlockingObjectPool.java

```diff
@@ -0,0 +1,57 @@
+package com.example.concurrency;
+
+import jakarta.annotation.PostConstruct;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Slf4j
+@Service
+public class BlockingObjectPool {
+
+    private List<Object> pool;
+    private final Object lock = new Object();
+
+    @Getter
+    private final int size = 10;
+
+    @PostConstruct
```

Comment:
Try to refrain to use such annotations. Prefer to init all stuff int the constructor. Or in case of more heavy initialization try to offload this process from bean initialization stage.
E.g. use lazy init or async, deferred init etc